packages/services/src/contents/index.ts

```diff
@@ -1080,6 +1099,28 @@ export class Drive implements Contents.IDrive {
     this._apiEndpoint = options.apiEndpoint ?? SERVICE_DRIVE_URL;
     this.serverSettings =
       options.serverSettings ?? ServerConnection.makeSettings();
+    this.contentProviderRegistry.drive = this;
+  }
+
+  static _contentProviderRegistry: IContentProviderRegistry | undefined =
+    undefined;
+
+  static getContentProviderRegistry(): IContentProviderRegistry {
+    if (Drive._contentProviderRegistry === undefined) {
+      const contentProviderRegistry = new ContentProviderRegistry();
+      const restContentProvider = new RestContentProvider();
+      contentProviderRegistry.register(restContentProvider);
+      Drive._contentProviderRegistry = contentProviderRegistry;
+    }
+    return Drive._contentProviderRegistry;
+  }
```

Comment:
Shall we move it to the non-exported `Private` namespace down in the file to match the existing code and ensure this is actually private?

Reply:
Wouldn't making `_contentProviderRegistry` private enough? `getContentProviderRegistry` must be public, so that extensions can register new content providers.
```ts
private static _contentProviderRegistry: IContentProviderRegistry | undefined = undefined;
```

Reply:
There is no `private static` in TypeScript, that's why we use `Private` non-exported namespaces :)

Reply:
Sorry, I was confused on `private static` availability. I guess that works too :+1: